update hibp to use real range API

groovecoder opened the pull request:
* Adds `HIBP_STAGE` env vars
* Adds handlebars template helpers to format the real HIBP breach metadata
* Updates `hibp` module to call the real stage endpoint for email hash range queries
* Removes `get-hashsets` script as we won't use it anymore

pdehaan commented on hbs-helpers.js:
```diff
@@ -0,0 +1,26 @@
+const HBSHelpers = {
+  init(hbs) {
+
+    hbs.registerHelper("prettyDate", function(date, options) {
```
Me thinks `options` is unused.
This also feels a bit too locale specific. What about something like `toLocaleDateString()`?

```js
return new Date(date).toLocaleDateString('en-US', {year: 'numeric', month: 'long', day: 'numeric'});
```

pdehaan commented on hibp.js:
```diff
@@ -17,25 +17,23 @@ const HIBP = {
     const sha1 = getSha1(email);
     const sha1Prefix = sha1.slice(0, 6);
-    const url = `${AppConstants.HIBP_STUB_API_ROOT}/breachedaccount/range/${sha1Prefix}`;
+    const url = `${AppConstants.HIBP_STAGE_API_ROOT}/range/${sha1Prefix}?code=${AppConstants.HIBP_STAGE_API_TOKEN}`;
```
I'm guessing the `HIBP_STAGE_API_TOKEN` is just a basic alpha-numeric string and doesn't need to be special URL-encoded?

groovecoder commented:
Addressed comments, and added 0d99beb7bfaf1943ebe5aad8ed91ecfcdb8694e3 to replace the last use of `popsicle` with `got` so we're using `got` everywhere. (I removed the last use of `request` when I removed the `get-hashsets` script.)

nhnt11 commented on package.json:
```diff
@@ -22,8 +21,7 @@
     "pg": "^7.4.3",
-    "popsicle": "^9.2.0",
-    "request": "^2.86.0"
+    "popsicle": "^9.2.0"
```
Can we get rid of popsicle from package.json now?

groovecoder replied:
oh, yeah sorry I meant to do that.

pdehaan commented on views/scan.hbs:
```diff
@@ -63,10 +63,11 @@
+            <span class="bold">Compromised data:</span> {{breachDataClasses meta.DataClasses }}<br>
+            <span class="bold">Compromised accounts:</span> {{localeString meta.PwnCount }}<br>
+            {{! TODO: make sure meta.Description is a safe string so we can use triple-{{{ }}
+            {{{ meta.Description }}}
```
Should we use something like [**dompurify**](https://github.com/cure53/DOMPurify) to sanitize the `meta.Description`?

groovecoder replied:
We should. Should it be part of this PR or a new issue?

pdehaan replied:
I'll file a new issue, but we should probably address before release.

groovecoder commented:
Merging. I'll fix up stage server when it deploys. It will need new required env vars.
